### PR TITLE
data-menu: UI tweaks

### PR DESCRIPTION
**Pull request by kecnry**
### Description

This pull request implements UI tweaks to the data-menu and API hints (app-wide) after consulting with @Jenneh.  These changes include:
* adjust color on orientation dropdown API hint to have more contrast with blue background
* show quotes and brackets around API hints for select dropdowns app-wide.  This resulted in creating two new reusable components - `plugin-select` and `glue-state-select`.  The goal in general is to make the API hints as directly "copy-pasteable" as possible.  Additional work can be done to text input fields in the future and maybe even a button or keyboard shortcut to copy the full code snippet to clipboard while hovering over an element with an API hint.
* margin on layer selection to make it more obvious when all are selected (addresses comment by @bmorris3: https://github.com/spacetelescope/jdaviz/pull/3254#issuecomment-2447947318)
* remove "..." on tooltips which otherwise appear as if they were truncated
* additional heading on header when orientation is present
* move API hint to access plugin API with new description (not shown below)
* custom icon for API hint toggle (not shown below)

To enable for testing (note: both this and the `_obj` will be removed when making the data-menu public):

```
for viewer in viz.viewers.values():
    viewer._obj.data_menu._obj.dev_data_menu = True
```

**Before**:
Screenshot 2024-11-20 at 8 23 47 AM

**After**:
Screenshot 2024-11-20 at 8 23 25 AM

### Change log entry

- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [ ] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [x] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone. Bugfix milestone also needs an accompanying backport label.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

**Comment by bmorris3** [comment]
Minor Q: if you adjust the orientation selection, the orientation dropdown+label get the accent color as though they're in focus/selected, until you click another item in the data menu: 

https://github.com/user-attachments/assets/20292bd5-0ef0-4864-817f-8a0fb57db478

Should it do that?

**Comment by bmorris3** [comment]
(This comment shouldn't block this PR/ticket.)

In developing the orientation plugin, we decided to show the "preset" orientation layers in the data menu _only_ after they've been created by clicking on the preset buttons in the plugin:

https://github.com/user-attachments/assets/22457554-cef5-4fb8-8428-9a78e11e5533

We _could_ expose each preset as an option in the new data menu, and only after each option is selected would we could create the corresponding WCS-only layer, load it into the viewer, and link it. Pros: this would save the users some clicks and scrolling to the orientation plugin. Cons: this would make it less clear which selectable options in the data menu are already-loaded data entries and which ones could be created on selection.

File: jdaviz/components/glue_state_select.vue
Comment (bmorris3) [inline jdaviz/components/glue_state_select.vue:27]:
```suggestion
  props: ['items', 'selected', 'label', 'api_hint', 'api_hints_enabled']
```

File: jdaviz/components/plugin_select.vue
Comment (bmorris3) [inline jdaviz/components/plugin_select.vue:73]:
```suggestion
          'api_hint', 'api_hints_enabled', 'dense', 'disabled']
```

File: jdaviz/configs/default/plugins/data_menu/data_menu.py
Comment (bmorris3) [inline jdaviz/configs/default/plugins/data_menu/data_menu.py:1]: Just writing in to say I appreciate these changes.

File: jdaviz/configs/default/plugins/plot_options/plot_options.vue
Comment (bmorris3) [inline jdaviz/configs/default/plugins/plot_options/plot_options.vue:189]: Impressive, I never would have caught a matplotlibian-slip like this.

File: jdaviz/configs/specviz2d/plugins/spectral_extraction/spectral_extraction.vue
Comment (bmorris3) [inline jdaviz/configs/specviz2d/plugins/spectral_extraction/spectral_extraction.vue:195]: Again, jeez.

**Review by bmorris3 — APPROVED** [review]
This all looks good, so I'm leaving an approval. I left a few videos above with behavior that should be fixed in this epic, but not necessarily this PR.

File: jdaviz/configs/default/plugins/plot_options/plot_options.vue
Comment (kecnry) [reply jdaviz/configs/default/plugins/plot_options/plot_options.vue:189]: Cami actually had during a demo, so I made sure to do a full search for `plt.` in the code 😂

**Comment by kecnry** [comment]
remaining TODO before merge:
- [ ] update API hints icon from @Jenneh with styling of "lock" button to match toolbars in plugins (semi-transparent orange)
- [ ] try to address https://github.com/spacetelescope/jdaviz/pull/3310#issuecomment-2494741175 - it seems that the `primary--text` class is being added by vue, and we need to prevent that from changing color or use something that looks better.  This color is coming from vue setting to a blue which is then inverted by us because its on a dark toolbar.
- [ ] update screenshots in PR description

**Review by gibsongreen — APPROVED** [review]
The updates look great here, I like the API icon and changes to the lock icon, the boldening with data items selections, dark-mode, and so on. Did see the known subset issue(s) but keeping that out of this review for later efforts. There were two things I noticed in testing, the second is likely out of scope and just an idea to reduce user-click.

1. When linked by pixel, should the Orientation label be above the button?

Screenshot 2024-11-29 at 10 51 31 AM


2. If linked by pixel, and the data loaded has WCS, would it be too much real-estate to add another button next to the existing one for the option to WCS link in the data menu or do we want to enforce user actions like this through the plugin itself?

**Comment by kecnry** [comment]
@gibsongreen I implemented your first suggestion - let's hold off on the second to see if there is demand for it and design it.  I think there are competing desires for convenience but also not having multiple places/ways to do the same thing.

Screenshot 2024-12-02 at 8 51 49 AM

**Pull request merged**
